go/vt/vtgate/grpcvtgateservice/server.go

```diff
@@ -168,6 +168,8 @@ func (vtg *VTGate) StreamExecute(request *vtgatepb.StreamExecuteRequest, stream
 		session.Options = request.Options
 	}
 	vtgErr := vtg.server.StreamExecute(ctx, session, request.Query.Sql, request.Query.BindVariables, func(value *sqltypes.Result) error {
+		// Send is not safe to call concurrently, but we
+		// guarantee that it's not.
```

Comment:
here and the same question for vtgate.go: Is it easy to figure out how we guarantee it? Should we document this more explicitly or is this fine? Up to you. I'm okay with the current version.